dvc/hash_info.py

```diff
@@ -0,0 +1,10 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class HashInfo:
```

Comment:
Loved this. I was having a lot of issues related to hashes when working with subrepos, mostly my fault but would have made debugging way easier.

Reply:
Also, a few `property`s would make it even better. Eg: `is_dir_checksum`?  But looks like it's not needed right now.

Reply:
We should try doing this more in the _internals_ instead of plain dict/tuples/list. Not saying we should use dataclass, but we can use `typing.NamedTuple/typing.TypedDict/dataclasses`. :slightly_smiling_face: 

The one that I can remember right now is `StatusInfo`. :)

Reply:
> Also, a few propertys would make it even better. Eg: is_dir_checksum? But looks like it's not needed right now.

Exactly! :slightly_smiling_face: Yeah, just don't want to push everything in this PR. Plus `is_dir` actually will require some stuff on index/tree side, so don't want to mix those up. Clearly it will be implemented in the near future though :slightly_smiling_face:

Reply:
> We should try doing this more in the internals instead of plain dict/tuples/list. Not saying we should use dataclass, but we can use typing.NamedTuple/typing.TypedDict/dataclasses/objects. slightly_smiling_face

Sorry, didn't get your point.

Reply:
> Sorry, didn't get your point.

Just trying to echo your point, that we should do this more instead of using plain dict/tuples. :arrow_down: 

> avoid using explicit tuples